scripts/gen_app_partitions.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+import sys
+import argparse
+import os
+import re
+import string
+from elf_helper import ElfHelper
+from elftools.elf.elffile import ELFFile
+
+
+# This script will create linker comands for power of two aligned MPU
+# when APP_SHARED_MEM is enabled.
+print_template = """
+/* Auto generated code do not modify */
+. = ALIGN( 1 << LOG2CEIL(data_smem_REPLACE_STRINGb_end - data_smem_REPLACE_STRING));
+data_smem_REPLACE_STRING = .;
+KEEP(*(SORT(data_smem_REPLACE_STRING*)))
+. = ALIGN(_app_data_align);
+data_smem_REPLACE_STRINGb_end = .;
+. = ALIGN( 1 << LOG2CEIL(data_smem_REPLACE_STRINGb_end - data_smem_REPLACE_STRING));
+"""
+
+full_list_of_partitions = []
+partitions_source_file= {}
```

Comment:
The use of global mutable variables like these two should be avoided.

Reply:
In an 86-line script?  Meh.  I'm with Adithya on this one.